src/components/Pagination.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from 'react';
+import './Pagination.scss';
+
+type Props = {
+  total: number;
+  perPage?: number;
+  page?: number;
+  onPageChange: (index: number) => void;
+  prevPage: () => void;
+  nextPage: () => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage = 5,
+  page = 1,
+  onPageChange,
+  prevPage,
+  nextPage,
+}) => {
+  const [isPrevDisabled, setIsPrevDisabled] = useState(true);
+  const [isNextDisabled, setIsNextDisabled] = useState(false);
+  const current = page;
+  const last = Math.ceil(total / perPage);
+
+  const paginate = (currentPage: number, lastPage: number) => {
+    const result = [];
+
+    for (
+      let i = Math.max(2, (currentPage - 1));
+      i <= Math.min((lastPage - 1), (currentPage + 1));
+      i += 1
+    ) {
+      result.push(i);
+    }
+
+    if ((currentPage - 1) > 2) {
+      result.unshift('...');
+    }
+
+    if ((currentPage + 1) < (lastPage - 1)) {
+      result.push('...');
+    }
+
+    result.unshift(1);
+    if (lastPage !== 1) {
+      result.push(lastPage);
```

Comment:
You can also try to use slice method for getting needed number of content from the content array